internal/database/testing/database.go

```diff
@@ -51,3 +52,60 @@ func DumpTable(c *gc.C, db *sql.DB, table string) {
 	writer.Flush()
 	fmt.Fprintln(os.Stdout)
 }
+
+// DumpTablesJSON prints the named tables to stdout in a JSON format. This is
+// intended as a debugging tool for state tests. It is not intended for use in
+// production code.
+func DumpTablesJSON(c *gc.C, db *sql.DB, tableNames ...string) {
```

Comment:
I don't get why we have both, one that's JSON and one that is tabwriter. I have a preference for the latter as it's much easier to grok at a glance.

Reply:
This is how the output of `DumpTable` displays in my terminal. Hopefully it illustrates why I'm not a fan.
```
uuid                                    activated       cloud_uuid   c
loud_region_uuid                        cloud_credential_uuid        m
odel_type_id    life_id name            owner_uuid                   
2885297f-46ee-48ca-84ea-28589de8f18a    true            1bc930de-8161-
4a2f-8a95-d779d01cbe60  aeb2a28c-6248-4931-88a0-94d79bfc415d    ecfece
dc-717b-4bba-8be6-408c6344b690  0               0       my-test-model4
f821fc6-337d-4536-875a-35e6dbbcb98c     974c3956-31b6-4031-8c73-bf4c18
588eab  true    1bc930de-8161-4a2f-8a95-d779d01cbe60    aeb2a28c-6248-
4931-88a0-94d79bfc415d  ecfecedc-717b-4bba-8be6-408c6344b690    0    0
listtest1       4f821fc6-337d-4536-875a-35e6dbbcb98c    b13a4188-3b62-
4e63-880c-2dedb5d1e1fe  true    1bc930de-8161-4a2f-8a95-d779d01cbe60 a
eb2a28c-6248-4931-88a0-94d79bfc415d     ecfecedc-717b-4bba-8be6-408c63
44b690  0       0       listtest2       4f821fc6-337d-4536-875a-35e6db
bcb98c
```

> I have a preference for the latter

Great, and I have a preference for the JSON. So surely there should be some room for personal preference in these utilities?

Reply:
Then pass in a formatter that allows you to express that.

```go
func DumpTablesJSON() { ... }
func DumpTables() { ... }

func dumpTables(..., formatted dumpTablesFormatter)
```